community/kernel/src/main/java/org/neo4j/kernel/impl/storageengine/impl/recordstorage/RecordStorageEngine.java

```diff
@@ -275,12 +272,6 @@ public StoreReadLayer storeReadLayer()
         return storeLayer;
     }
 
-    @Override
-    public CommandReaderFactory commandReaderFactory()
-    {
-        return commandReaderFactory;
```

Comment:
Removing this is counter to how it was intended. A `CommandReaderFactory` is designed to be state-less and so it should be enough with one per database instance. It may be that other places who currently instantiates a `RecordStorageCommandReaderFactory` should really be getting this instance instead, especially hot paths that read commands. I'd like that way better than to remove this method.

`VersionAwareLogEntryReader` has a default constructor which creates a new `RecordStorageCommandReaderFactory` mostly for convenience in tests. Now I see that it has been used and new instances of these readers are created in many places, even if it should always be dependent on which specific storage engine is used underneath. It's fine currently, but will obviously break if/when in the future having multiple storage engines.

You don't have to do anything to other parts of the code, but would you mind keeping this method at least (and I'll create a card for the other stuff)?

Reply:
The main problem is that the `LogEntryReader` now needs to be created before the `RecordStorageCommandReaderFactory`, I'll restore the functions and add the card to the backlog for now. The `VersionAwareLogEntryReader` is only created once now in the hot path, so it's not a huge problem.